src/Core/MySQL/MySQLReplication.cpp

```diff
@@ -852,68 +853,74 @@ namespace MySQLReplication
                 break;
             }
-            case TABLE_MAP_EVENT: {
+            case TABLE_MAP_EVENT:
+            {
                 event = std::make_shared<TableMapEvent>();
-                event->parseHeader(payload);
-                event->parseEvent(payload);
+                event->parseHeader(event_payload);
+                event->parseEvent(event_payload);
                 table_map = std::static_pointer_cast<TableMapEvent>(event);
                 break;
             }
             case WRITE_ROWS_EVENT_V1:
-            case WRITE_ROWS_EVENT_V2: {
+            case WRITE_ROWS_EVENT_V2:
+            {
                 if (do_replicate())
                     event = std::make_shared<WriteRowsEvent>(table_map);
                 else
                     event = std::make_shared<DryRunEvent>();
 
-                event->parseHeader(payload);
-                event->parseEvent(payload);
+                event->parseHeader(event_payload);
+                event->parseEvent(event_payload);
                 break;
             }
             case DELETE_ROWS_EVENT_V1:
-            case DELETE_ROWS_EVENT_V2: {
+            case DELETE_ROWS_EVENT_V2:
+            {
                 if (do_replicate())
                     event = std::make_shared<DeleteRowsEvent>(table_map);
                 else
                     event = std::make_shared<DryRunEvent>();
 
-                event->parseHeader(payload);
-                event->parseEvent(payload);
+                event->parseHeader(event_payload);
+                event->parseEvent(event_payload);
                 break;
             }
             case UPDATE_ROWS_EVENT_V1:
-            case UPDATE_ROWS_EVENT_V2: {
+            case UPDATE_ROWS_EVENT_V2:
+            {
                 if (do_replicate())
                     event = std::make_shared<UpdateRowsEvent>(table_map);
                 else
                     event = std::make_shared<DryRunEvent>();
 
-                event->parseHeader(payload);
-                event->parseEvent(payload);
+                event->parseHeader(event_payload);
+                event->parseEvent(event_payload);
                 break;
             }
-            case GTID_EVENT: {
+            case GTID_EVENT:
+            {
                 event = std::make_shared<GTIDEvent>();
-                event->parseHeader(payload);
-                event->parseEvent(payload);
+                event->parseHeader(event_payload);
+                event->parseEvent(event_payload);
                 position.update(event);
                 break;
             }
-            default: {
+            default:
+            {
                 event = std::make_shared<DryRunEvent>();
-                event->parseHeader(payload);
-                event->parseEvent(payload);
+                event->parseHeader(event_payload);
+                event->parseEvent(event_payload);
                 break;
             }
         }
-        payload.ignoreAll();
```

Comment:
Why `payload.ignoreAll();` or `payload.ignore(CHECKSUM_CRC32_SIGNATURE_LENGTH);` doesn't work and we need `MySQLBinlogEventReadBuffer` to skip the last 4 bytes in packet?

Reply:
We need to extract this logic to accommodate the different checksum algorithms(https://dev.mysql.com/doc/refman/8.0/en/replication-options-binary-log.html#option_mysqld_binlog-checksum) and make it easier to use some read methods. for example: `readStringUntilEOF` and other readUntilEOF.

As a bonus, we don't need to care about checksum in event parsing, and the code is probably more readable.